_search-plugins/concurrent-segment-search.md

```diff
@@ -0,0 +1,162 @@
+
+- When sending long-running requests, for example, requests that contain aggregations or large ranges.
+- As an alternative to force-merging segments into a single segment in order to improve performance.
+
+## Background
+
+In OpenSearch, each search request follows the scatter-gather protocol. The coordinating node receives a search request, evaluates which shards are needed to serve this request, and sends a shard-level search request to each of those shards. Each shard that receives the request executes the request locally using Lucene and returns the results. The coordinating node merges the responses received from all shards and sends the search response back to the client. Optionally, the coordinating node can perform a fetch phase before returning the final results to the client if any document field or the entire document is requested by the client as part of the response.
+
+## Searching segments concurrently
+
+Without concurrent segment search, Lucene executes a request sequentially across all segments on each shard during query phase. The query phase then collects the top hits for the search request. With concurrent segment search, each shard-level request will search the segments in parallel during query phase. For each shard, the segments are divided into multiple _slices_. Each _slice_ is the unit of work that can be executed in parallel on a separate thread, and thus, the slice count determines the maximum degree of parallelism for a shard-level request. Once all the slices complete their work, Lucene performs a reduce operation on the slices, merging them and creating the final result for this shard-level request. Slices are executed using a new `index_searcher` thread pool, which is different from the `search` thread pool that handles shard-level requests.
+
+## Enabling the feature flag
+
+There are several methods for enabling concurrent segment search, depending on the install type. 
+
+### Enable in opensearch.yml
+
+If you are running an OpenSearch cluster and want to enable concurrent segment search in the config file, add the following line to `opensearch.yml`:
+
+```yaml
+opensearch.experimental.feature.concurrent_segment_search.enabled=true
+```
+{% include copy.html %}
+
+### Enable with Docker containers
+
+If you’re running Docker, add the following line to `docker-compose.yml` under the `opensearch-node` > `environment` section:
+
+```bash
+OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true"
+```
+{% include copy.html %}
+
+### Enable on a node using a tarball install
+
+To enable concurrent segment search on a tarball install, provide the new JVM parameter either in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS`.
+
+#### OPTION 1: Modify jvm.options
+
+Add the following lines to `config/jvm.options` before starting the OpenSearch process to enable the feature and its dependency:
```

Comment:
Can we be a little more precise than "the OpenSearch process"? Which one, specifically?

Reply:
The literal process called "opensearch". I'll put it in code format